### DISCUSSION TITLE
How to export struct member as rust type Vec<XXX> but not ModeRC<XXX> in slint ?

### DISCUSSION THREAD

**Discussion by fluolite** [slint-ui/slint#6950] [General]
Such as the following code.

# slint code  
```
export struct DeviceGroupModel {
    group_id: int,
    groups: [DeviceGroupVo],
}

```

**Comment by hunger** [comment]
It's a `ModelRc<T>` since we need change notification.

You can extract the information into a `Vec<T>` in the business logic when needed. If you use a `VecModel` to back the `ModelRc`, then that should be reasonably cheap (but includes a clone!). A `SharedVecModel` can get you a slice, so that could avoid the copy call, but at the price of not having a real `Vec`.

To downcast from a `ModelRc` to a `VecModel`, use:

```rust
model_rc.as_any().downcast_ref::<VecModel<i32>>().unwrap().push(4);
```
